File: apm-method-plugin/src/main/groovy/com/will/method/plugin/MethodPlugin.java
package com.will.method.plugin;

import com.android.build.gradle.AppExtension;
import com.android.build.gradle.internal.LoggerWrapper;
import com.will.base.utils.Log;
import com.will.method.plugin.bytecode.DebugWeaver;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

import java.util.Collections;

public class MethodPlugin implements Plugin<Project> {
    private static final LoggerWrapper logger = LoggerWrapper.getLogger(DebugWeaver.class);
    @SuppressWarnings("NullableProblems")
    @Override
    public void apply(Project project) {

        Log.d("","**********************************");
        Log.d("","*******                    *******");
        Log.d("","*******       plugin       *******");
        Log.d("","*******                    *******");
        Log.d("","**********************************");
        AppExtension appExtension = (AppExtension)project.getProperties().get("android");
        appExtension.registerTransform(new MethodTransform(project), Collections.EMPTY_LIST);
    }

}

File: apm-method-plugin/src/main/kotlin/com/will/method/plugin/MethodPlugin.kt
package com.will.method.plugin

import com.android.build.gradle.AppExtension
import com.will.base.utils.Log

import org.gradle.api.Plugin
import org.gradle.api.Project

import java.util.Collections

class MethodPlugin : Plugin<Project> {
    override fun apply(project: Project) {

        Log.d("", "**********************************")
        Log.d("", "*******                    *******")
        Log.d("", "*******       plugin       *******")
        Log.d("", "*******                    *******")
        Log.d("", "**********************************")
        val appExtension = project.properties["android"] as AppExtension
        appExtension.registerTransform(MethodTransform(project), Collections.EMPTY_LIST)
    }
}